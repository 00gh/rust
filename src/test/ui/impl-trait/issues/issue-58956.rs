trait Lam {}

pub struct B;
impl Lam for B {}
pub struct Wrap<T>(T);

const _A: impl Lam = {
    //~^ `impl Trait` not allowed outside of function and inherent method return types
    let x: Wrap<impl Lam> = Wrap(B);
    //~^ `impl Trait` not allowed outside of function and inherent method return types
    x.0
};

fn main() {}
