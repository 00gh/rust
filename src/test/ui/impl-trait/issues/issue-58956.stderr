error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/issue-58956.rs:7:11
   |
LL | const _A: impl Lam = {
   |           ^^^^^^^^
   |
   = note: found `impl Trait` in type

error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> $DIR/issue-58956.rs:9:17
   |
LL |     let x: Wrap<impl Lam> = Wrap(B);
   |                 ^^^^^^^^
   |
   = note: found `impl Trait` in variable

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0562`.
